src/lib/services/feature-toggle-service.ts

```diff
@@ -1247,9 +1248,26 @@ class FeatureToggleService {
         newVariants: Operation[],
         createdBy: string,
     ): Promise<FeatureToggle> {
-        const oldVariants = await this.getVariants(featureName);
-        const { newDocument } = applyPatch(oldVariants, newVariants);
-        return this.saveVariants(featureName, project, newDocument, createdBy);
```

Comment:
`saveVariants` iterates over all feature_environments overriding the variants in all environments which was causing this issue